### PR TITLE
update dev dependencies, husky, prettier, eslint configs

### PR DESCRIPTION
**Pull request by dai-shi**
to be sync as much as possible with the valtio repo.

### CODE REVIEW

**Comment by barelyhuman** [comment]
Mostly done, 
One thing pending is moving from babel-eslint to @babel/eslint-parser, 
the newer `@babel/eslint-parser` might not be in sync with the existing checks for mutating the proxy state, which I'll be checking later tomorrow, 

please avoid making a release till then, there's typo in the PR #20 , for one of the lines, while the test cases worked, there's still an issue, which is fixed in this

**Comment by barelyhuman** [comment]
@dai-shi mostly done, would want you to test the CSB package once before we merge, I'm also doing it

**Comment by dai-shi** [comment]
Would it be possible to make the package json file as close as possible to the one in the valtio repo? Especially the order of fields (because we use `diff` to see differences). Not crucial but seems consistent and easy to follow.

**Comment by barelyhuman** [comment]
Will be done 👍🏼

File: package.json [hunk 1]
```diff
@@ -4,35 +4,37 @@
...
   "scripts": {
+    "prepare": "husky install",
     "prebuild": "shx rm -rf dist",
     "build": "rollup -c",
     "postbuild": "yarn copy",
-    "eslint": "eslint --fix '{src,tests}/**/*.{js,ts,jsx,tsx}'",
-    "eslint:ci": "eslint '{src,tests}/**/*.{js,ts,jsx,tsx}'",
-    "prepare": "yarn build",
+    "eslint": "eslint --fix '{src,tests}/**/*.{js,jsx}'",
+    "eslint:ci": "eslint '{src,tests}/**/*.{js,jsx}'",
```
Comment (dai-shi) [inline package.json:16]: Hoped to check `'*.{js,json}'` too but something is blocking?

File: src/StateSnapshot.js [hunk 2]
```diff
@@ -114,6 +114,7 @@ function outerMemberExpression(node) {
   }
   return outerMemberExpression(node.parent)
 }
+// eslint-disable-next-line
```
Comment (dai-shi) [inline src/StateSnapshot.js:117]: can you specify which rule to disable?

**Review by dai-shi — COMMENTED** [review]
Looks good. Thanks for your work!

File: package.json (hunk 1 above)
Comment (barelyhuman) [reply package.json:16]: I wasn't able to fix this and nor was I able to find documentation regarding it. 
Eslint was trying to validate it's own config and `package.json` but then expecting semi-colon's in json files. Which didn't make sense, there's not much about this issue anywhere, I tried downgrading but no luck

Comment (dai-shi) [reply package.json:16]: okay, let's leave it. thanks.

File: src/StateSnapshot.js (hunk 2 above)
Comment (barelyhuman) [reply src/StateSnapshot.js]: done

**Pull request merged**